src/v/serde/json/detail/tests/string_test.cc

```diff
@@ -50,8 +50,9 @@ struct test_case {
 };
 
 std::ostream& operator<<(std::ostream& os, const test_case& tc) {
-    os << "input: " << tc.input
-       << ", expected_err: " << static_cast<int>(tc.expected_err)
+    // Don't print the input string as it's generally not valid UTF-8
+    // and it breaks test reports.
```

Comment:
fwiw, i think there are utilities out there that will replace invalid utf-8 code points with some valid filler, or escape them somehow. but this seems simple enough too.

Reply:
i've used something like that here https://github.com/redpanda-data/redpanda/blob/3915ed2e256d8a14cb0a1227e8e95c0744dbe606/src/v/serde/json/tests/dom.cc#L24